projects/vdk-plugins/vdk-impala/src/vdk/plugin/impala/lineage_utils.py

```diff
@@ -0,0 +1,55 @@
+# Copyright 2021 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+import logging
+import re
+from typing import Optional
+from typing import Tuple
+
+from impala._thrift_gen.RuntimeProfile.ttypes import TRuntimeProfileFormat
+from impala.hiveserver2 import HiveServer2Cursor
+from vdk.api.lineage.model.sql.model import LineageData
+from vdk.api.lineage.model.sql.model import LineageTable
+
+_log = logging.getLogger(__name__)
+
+
+def get_lineage_data(cursor: HiveServer2Cursor) -> Optional[LineageData]:
+    """
+    Calculates lineage information from the query plan for the last executed query from the provided cursor. Lineage
+    is calculated only if the query has finished successfully and has resulted actual data read or write from/to the
+    underlying storage.
+    """
+    query_profile = cursor.get_profile(profile_format=TRuntimeProfileFormat.STRING)
```

Comment:
So, my understanding is the driver is doing `select 1` in the background, in the scope of the active session.
At the same time, `profile` is using the latest query metadata.
Is there a possibility, that the latest query is in fact `select 1`, and what happens in such a case?

Reply:
Although I have not seen any official documentation, my observations are that select 1 queries are executed right before the actual query is executed. Moreover we are hooking before the result set is consumed, so there should not be any other query executed in the mean time.